docs/DeveloperGuide.md

```diff
@@ -1,34 +1,890 @@
+and enters a switch case for execution.  
+3. `parseUserCommand` in the Parser object will call a method `SaveStateCommand`.  
+4. On execute(), `Storage.copyFile()` will be called three times to copy contents of ingredients, recipes and chore save files into their respective manual-mode save files.
+
+##### Design considerations:
+Aspects: How saving of current state data executes:
+
+- Alternative 1 (current choice): Using Files.copy to copy content of auto-save files to manual-save files.
+
+|     |     |
+|-----|-----|
+|**Pros** | Easy to implement as less code is needed with Java 7 Files helper class.|  
+|**Cons** | Relatively slow copy performance when file size increases.|
+
+- Alternative 2: Using FileChannels to copy content of auto-save files to manual-save files.
+
+|     |     |
+|-----|-----|
+|**Pros** | The FileChannels technique is usually faster than its alternatives such as basic streams.|  
+|**Cons** | It may fail for very large files and more lines of codes are needed for implementation.|
+
+
+### 4.5. Logging
+Logging in the application refers to storing exceptions, warnings and messages that occur during the execution of Kitchen Helper. It was included to help developers to identify bugs and to simplify their debugging process. 
+
+The `java.util.logging` package in Java is used for logging. The logging mechanism can be managed from the `KitchenHelper` class through the `kitchenLogs` logger object.
+
+All control of the logger for the application can be viewed/ altered in the `setUpLogger()` method. The current settings for the logger are as follow:
+
+- All logs of `Level.SEVERE` level will be shown on the console when an input/ program flow has caused a possible disruption to the execution of the program. (See the levels of logging below)
+- All information above ‘Level.FINE’ level is logged into a log file, `KitchenLogs.log`.
+- Logging is made to be displayed in the `SimpleFormatter` style where the date, class and error description are logged.
+
+Logging Levels:
+- `Level.SEVERE`: a serious failure, which prevents normal execution of the program, for end users and system administrators.
+- `Level.WARNING`: a potential problem, for end users and system administrators.
+- `Level.INFO`: reasonably significant informational message for end users and system administrators.
+- `Level.CONFIG`: hardware configuration, such as CPU type.
+- `Level.FINE`, `Level.FINER`, `Level.FINEST`: three levels used for providing tracing information for the software developers.
+
+Additional logging can be done by adding the calling of the global logger and invoking the function `log()`. This will ensure that all loggings will be made to the same file across the various classes. 
+
+An example is shown below:
+```java
+public static final Logger kitchenLogs = Logger.getLogger(Logger.GLOBAL_LOGGER_NAME);
+kitchenLogs.log(Level.WARNING, description_of_warning_here, e.toString());
+```
+
+### 4.6. Configuration
```

Comment:
Was there something that should be under configuration? Rmb to complete it and all the best! :)